includes/Core/Util/Migration_1_8_1.php

```diff
@@ -0,0 +1,257 @@
+		$this->authentication = $authentication ?: new Authentication( $this->context, $this->options, $this->user_options );
+	}
+
+	/**
+	 * Registers hooks.
+	 *
+	 * @since 1.8.1
+	 */
+	public function register() {
+		add_action( 'admin_init', array( $this, 'migrate' ) );
+	}
+
+	/**
+	 * Migrates the DB.
+	 *
+	 * @since 1.8.1
+	 */
+	public function migrate() {
+		$db_version = $this->options->get( 'googlesitekit_db_version' );
+
+		// Do not run if database version already updated.
+		if ( $db_version && version_compare( $db_version, self::DB_VERSION, '>=' ) ) {
+			return;
+		}
+
+		// Only run routine if using the authentication service, otherwise it
+		// is irrelevant.
+		if ( ! $this->authentication->get_oauth_client()->using_proxy() ) {
+			return;
+		}
+
+		// Only run routine once site credentials present, otherwise it is not
+		// possible to connect to the authentication service.
+		if ( ! $this->authentication->credentials()->has() ) {
+			return;
+		}
+
+		$this->clear_and_flag_unauthorized_verified_users();
+
+		// Update database version.
+		$this->options->set( 'googlesitekit_db_version', self::DB_VERSION );
+	}
+
+	/**
+	 * Checks whether there are any users that are verified without proper
+	 * authorization, clear their Site Kit data, and flag them on the
+	 * authentication service.
+	 *
+	 * @since 1.8.1
+	 *
+	 * @return boolean|WP_Error True on success, WP_Error on failure.
+	 */
+	private function clear_and_flag_unauthorized_verified_users() {
+		// Detect all unauthorized verified users and clean their Site Kit data.
+		$unauthorized_identifiers = $this->clear_unauthorized_verified_users();
+
+		// If no unauthorized verified users found, all is well, no need to
+		// show a notification.
+		if ( empty( $unauthorized_identifiers ) ) {
+			return true;
+		}
+
+		// Flag site as affected so that the notification to inform and explain
+		// steps to resolve will be shown.
+		$credentials  = $this->authentication->credentials()->get();
+		$google_proxy = new Google_Proxy( $this->context );
+		$response     = wp_remote_post(
+			$google_proxy->url( '/notifications/mark/' ),
+			array(
+				'body' => array(
+					'site_id'            => $credentials['oauth2_client_id'],
+					'site_secret'        => $credentials['oauth2_client_secret'],
+					'notification_id'    => 'verification_leak',
+					'notification_state' => 'required',
+					// This is a special parameter only supported for this
+					// particular notification.
+					'identifiers'        => implode( ',', $unauthorized_identifiers ),
+				),
+			)
+		);
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+		if ( 200 !== $response_code ) {
+			$body    = wp_remote_retrieve_body( $response );
+			$decoded = json_decode( $body, true );
+			return new WP_Error( $response_code, ! empty( $decoded['error'] ) ? $decoded['error'] : $body );
+		}
+		return true;
```

Comment:
`$response` and subsequent return values are no longer used and can be removed. That should also mean that we can set `blocking => false`.

Reply:
I think this method returning values is not problematic. I agree it's weird for an action hook callback method to return something, but arbitrary methods like this can certainly return values, which would have the benefit to make testing them easier (even if we don't have direct coverage here).